src/plugins/home/public/application/application.tsx

```diff
@@ -77,3 +77,28 @@ export const renderApp = async (
     unlisten();
   };
 };
+
+export const renderImportSampleDataApp = async (
+  element: HTMLElement,
+  coreStart: CoreStart,
+  history: ScopedHistory
+) => {
+  // dispatch synthetic hash change event to update hash history objects
+  // this is necessary because hash updates triggered by using popState won't trigger this event naturally.
+  // This must be called before the app is mounted to avoid call this after the redirect to default app logic kicks in
+  const unlisten = history.listen((location) => {
```

Comment:
Do we need this for import sample data app? It seems it's home page app specific.